File: leetcode/3372-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        if len(nums)==1:
            return 1
        countinc = 1
        m = 1
        for i in range(len(nums)-1):
            if nums[i]<nums[i+1]:
                countinc+=1
                m = max(m,countinc)
            else:
                countinc = 1

        
        countdec = 1
        for i in range(len(nums)-1):
            if nums[i]>nums[i+1]:
                countdec+=1
                m = max(m,countdec)
            else:
                countdec = 1
        
        return m

File: leetcode/3372-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        if (nums.size()==1){
            return 1;
        }
        int countinc = 1, countdec = 1, m = 1;

        for(int i = 0;i<nums.size()-1;i++){
            if(nums[i]<nums[i+1]){
                countinc++;
                m = max(m,countinc);
            }
            else{
                countinc = 1;
            }
        }
        for(int i = 0;i<nums.size()-1;i++){
            if(nums[i]>nums[i+1]){
                countdec++;
                m = max(m,countdec);
            }
            else{
                countdec = 1;
            }
        }
        return m;
    }
};
